ServiceFabric.PubSubActors/Stats/StatsCollector.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System.Threading.Tasks;
+using ServiceFabric.PubSubActors.State;
+
+namespace ServiceFabric.PubSubActors.Stats
+{
+    public class StatsCollector : IStatsCollector
+    {
+        public Task OnMessageReceived(ReferenceWrapper subscriber, MessageWrapper messageWrapper)
+        {
+            subscriber.TotalReceived++;
```

Comment:
This is exactly what I mean yes, except for the call to set the `TotalReceived` on the subscriber. I was thinking that this `StatsCollector` could be renamed into `InMemoryStatCollector` (which keeps track of delivered messages in memory) and that we could have alternate implementations, like one that sends a message into ETW and one that sends custom telemetry into AppInsights.